pkg/controller/kappnav/kappnav_controller.go

```diff
@@ -37,12 +38,13 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 	"sigs.k8s.io/yaml"
 )
 
-var log = logf.Log.WithName("controller_kappnav")
+// create new logger
+var	logger = kappnavutils.NewLogger(true)  //log in JSON format
```

Comment:
I believe we agreed that the operator should remain stateless. Given that the logging setting is tied to the kAppNav CR which is passed in as part of the context to the Reconcile() function, I would have expected the configured logger object to be local to the function (instead of a static variable) and passed as part of the context to other functions (e.g. utils).

Reply:
Other functions in kappnav_controller do not pass in kappnav CR and I can't log the message without creating NewLogger